clouddrift/wavelet.py

```diff
@@ -0,0 +1,801 @@
+"""
+This module provides wavelet functions.
+"""
+
+import numpy as np
+from typing import Optional, Tuple, Union
+import xarray as xr
+import warnings
+from scipy.special import gamma as _gamma, gammaln as _lgamma
+
+
+def morse_wavelet_transform(
+    x: np.ndarray,
+    gamma: float,
+    beta: float,
+    radian_frequency: np.ndarray,
+    complex: Optional[bool] = False,
+    order: Optional[int] = 1,
+    normalization: Optional[str] = "bandpass",
+    boundary: Optional[str] = "mirror",
+    time_axis: Optional[int] = -1,
+) -> Union[Tuple[np.ndarray], np.ndarray]:
+    """
+    Apply a continuous wavelet transform to an input signal using the generalized Morse
+    wavelets of Olhede and Walden (2002). The wavelet transform is normalized differently
+    for complex-valued input than for real-valued input, and this in turns depends on whether the
+    optional argument ``normalization`` is set to ``"bandpass"`` or ``"energy"`` normalizations.
+
+    Parameters
+    ----------
+    x : np.ndarray
+        Real- or complex-valued signals.
+    gamma: float
+       Gamma parameter of the Morse wavelets.
+    beta: float
+       Beta parameter of the Morse wavelets.
+    radian_frequency: np.ndarray
+       An array of radian frequencies at which the Fourier transform of the wavelets
+       reach their maximum amplitudes. ``radian_frequency`` is between 0 and 2 * np.pi * 0.5,
```

Comment:
should this range be validated?

Reply:
That should not be necessary because radian frequencies are cyclic, i.e pi = 3pi etc. I modified the docstring to state that radian frequency is typically between 0 and 2 np.pi * 0.5